crates/core/src/client.rs

```diff
@@ -279,6 +311,34 @@ async fn sn_state_root_inner(l1_client: &Client<impl Database>, contract_addr: A
     FieldElement::from_byte_slice_be(&starknet_root).map_err(|e| eyre!(e))
 }
 
+async fn wrapped_retry_sn_state_root_inner(
```

Comment:
why should we have impl`wrapped_retry_sn_state_root_inner` and `wrapped_retry_sn_state_block_number_inner` separeted?

Reply:
Fixed. They are merged into the `wrapped_retry` function. I will resolve this conversation but feel free to reopen it ;)